src/app/shared/components/theme-mode-toggle/theme-mode-toggle.component.ts

```diff
@@ -1,49 +1,57 @@
-import { Component, Inject, OnInit } from '@angular/core';
-import { DOCUMENT } from '@angular/common';
 import { MediaMatcher } from '@angular/cdk/layout';
+import { DOCUMENT } from '@angular/common';
+import { ChangeDetectorRef, Component, Inject, OnInit } from '@angular/core';
 import { StorageService } from '../../services/storage.service';
 
+type Theme = 'light' | 'dark';
+
 @Component({
   selector: 'app-theme-mode-toggle',
   templateUrl: './theme-mode-toggle.component.html',
   styleUrls: ['./theme-mode-toggle.component.scss'],
 })
 export class ThemeModeToggleComponent implements OnInit {
-  isDarkMode: boolean;
+  theme: Theme;
 
   constructor(
     @Inject(DOCUMENT)
     private readonly document: Document,
     private readonly mediaMatcher: MediaMatcher,
     private readonly storageService: StorageService,
+    private readonly changeDetector: ChangeDetectorRef,
   ) {}
 
   ngOnInit() {
-    // This is commented out because by default the theme mode is set to light (at least for now)
-    // const userPrefersTheme =
-    //   this.mediaMatcher.matchMedia &&
-    //   this.mediaMatcher.matchMedia('(prefers-color-scheme: light)').matches;
-    // this.setThemeMode(this.getUserSettingsIsDarkMode() || userPrefersTheme);
```

Comment:
Wouldn't just uncommenting this code be enough?

Reply:
This is what I originally hoped for, but it didn't seem to be working. So I started fixing it.

https://github.com/user-attachments/assets/45827489-7940-4393-99ef-d7c199e6b5f7

Reply:
Are you certain there's nothing in your local storage that specifies which mode to use? If I remember correctly, the commented-out code checks local storage first and, if it's empty, falls back to the user's preferred color scheme.

Reply:
> Are you certain there's nothing in your local storage that specifies which mode to use?

Yes, absolutely. You can see on the video that my `localStorage` is empty.

Moreover, looking at the old code, it isn't even supposed to listen to any changes in color scheme preferences, since there is no `onChange` event listener specified anywhere. It only supposed to get the system preferences on the initial load (`ngOnInit`), but it turns out that it also doesn't work, if we just uncomment the code. I get the light mode despite of my system preference being dark.

It might seem like the code on lines 28-29, which overrides everything returned by the `mediaMatches`, was put in there as a temporary substitution for the commented code. If we remove that and leave only uncommented code, it starts working but in the opposite direction: if my system preference is dark, it gives me the light mode and, vice versa, if my system preference is light, only then it gives me the dark mode.

However, if we change the `mediaMatcher` query from `(prefers-color-scheme: light)` to `(prefers-color-scheme: dark)`, it starts to give me the right theme.

But it still doesn't listen to any changes in the system preferences. If you change your system preference manually or if your OS configured to change the system preference automatically at different times of the day, you **have to** refresh your page in order for theme to change. This is what I intended to implement in this PR.

I added an event listener, which, if there's no stored theme in `localStorage`, would automatically change the theme according to the system preference, with no need to refresh the page. Also there's `skipStorage` parameter which allows skipping saving the theme in `localStorage`. We need this flag in order to prevent `onChange` event from storing anything in `localStorage`. Otherwise, once our system preference changes for the first time, the site will no longer respect it, despite the user not selecting the preferred theme manually by clicking the theme button. This is not the behavior we usually expect and I'd consider it a bug. So this is already handled.

What I'm not so sure about, though, is the use of `changeDetectorRef`. I don't know if this is a good or a bad practice, since this PR is basically the first time I ever touch Angular. So I would appreciate any advice regarding it.

What I've added it for is that when the system preference change, the theme icon in the top right corner stays the same. You can notice it on the video I first attached to the PR. I recorded it before noticing it myself and before I pushed the fix.

For some reason, when `theme` change was caused by `darkSchemeMatcher.onchange` event, `*ngIf`s in the `.html` file didn't react to the change and stayed the same. So I found this approach with `changeDetectorRef` and with it all went as expected.